internal/backend/state.go

```diff
@@ -507,3 +581,13 @@ func (state *State) close(ctx context.Context, tx *ent.Tx) error {
 
 	return nil
 }
+
+func canon(name, del string) string {
```

Comment:
This can be simplified to just check the first 5 characters/runes of the string and replace them with the uppercase version if they match the sequence `INBOX`. Alternatively, one could also do this in the parser and then you don't have check all the current and future cases.